### PR TITLE
Test with pydantic and add Union support

### PR DESCRIPTION
**Pull request by lorenzocelli**
I added a new test file `test_type_schema.py` to test the `TypeSchema` class using pydantic's `TypeAdapter`. With this changes tool2schema now supports nested generic types such as `List[Optional[Literal["a", "b", "c"]]]`.

### CODE REVIEW

File: tool2schema/type_schema.py [hunk 1]
```diff
@@ -96,9 +101,9 @@ def to_json(self) -> dict:
         Return the json schema for this type.
         """
         fields = {
-            "type": self._get_type(),
             "items": self._get_items(),
             "enum": self._get_enum(),
+            **self._get_type(),
```
Comment (lorenzocelli) [inline tool2schema/type_schema.py:106]: This is now inconsistent with the other two methods. I was thinking of having them return dictionaries as well, but then overriding methods would have to repeat the `"enum"`/`"items"` keyword. Not sure which way is better

File: tests/test_type_schema.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+def test_list_encode_decode_enum():
+    type_schema = TypeSchema.create(List[CustomEnum])
+
+    adapter = TypeAdapter(List[Literal["YES", "NO", "MAYBE"]])
+    assert type_schema.to_json() == adapter.json_schema()
+
+    assert type_schema.encode([CustomEnum.YES]) == ["YES"]
+    assert type_schema.decode([CustomEnum.YES]) == [CustomEnum.YES]
+    assert type_schema.decode(["YES"]) == [CustomEnum.YES]
```
Comment (siliconlad) [inline tests/test_type_schema.py:74]: Would add one decode and encode test with 2 elements.

File: tests/test_type_schema.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+def test_list_optional_encode_decode_enum():
+    type_schema = TypeSchema.create(List[Optional[CustomEnum]])
+
+    adapter = TypeAdapter(List[Optional[Literal["YES", "NO", "MAYBE"]]])
+    assert type_schema.to_json() == adapter.json_schema()
+
+    assert type_schema.encode([CustomEnum.YES]) == ["YES"]
+    assert type_schema.decode([CustomEnum.YES]) == [CustomEnum.YES]
+    assert type_schema.decode(["YES"]) == [CustomEnum.YES]
+    assert type_schema.decode([None]) == [None]
```
Comment (siliconlad) [inline tests/test_type_schema.py:86]: Would add one decode and encode test with 2 elements.

File: tool2schema/type_schema.py (hunk 1 above)
Comment (siliconlad) [reply tool2schema/type_schema.py:106]: This is probably fine for now.

File: tests/test_type_schema.py (hunk 3 above)
Comment (lorenzocelli) [reply tests/test_type_schema.py]: Parametrized with f0696e5

File: tests/test_type_schema.py (hunk 2 above)
Comment (lorenzocelli) [reply tests/test_type_schema.py]: Parametrized with 465da71

**Review by siliconlad — APPROVED** [review]
LGTM

**Pull request merged**